src/main/java/com/avispl/dal/avdevices/audionetworkinterface/studiotechnologies/danteleaderclock5401a/DanteLeaderClockCommunicator.java

```diff
@@ -119,82 +154,82 @@ public List<Statistics> getMultipleStatistics() throws Exception {
-			localExtendedStatistics.getStatistics().putAll(stats);
-			return Collections.singletonList(localExtendedStatistics);
-		}
-		if (sentRequests.get(DanteLeaderClockConstant.GENERAL).equals(false)) {
-			generalProperties(stats);
-			sentRequests.put(DanteLeaderClockConstant.GENERAL, true);
-			localExtendedStatistics.getStatistics().putAll(stats);
-			return Collections.singletonList(localExtendedStatistics);
+		failedMonitor.clear();
+		generalProperties(stats);
+		syncInputProperties(stats);
+		networkProperties(stats);
+		toneGeneratorProperties(stats);
+		systemProperties(stats);
+		if (failedMonitor.size() == DanteLeaderClockConstant.NO_OF_MONITORING_METRICS) {
+			StringBuilder stringBuilder = new StringBuilder();
+			stringBuilder.append(DanteLeaderClockConstant.FAIL_TO_GET_MONITORING_DATA);
+			for (Map.Entry<String, String> messageFailed : failedMonitor.entrySet()) {
+				String value = messageFailed.getValue();
+				if (value.contains(DanteLeaderClockConstant.FAIL_POPULATE_ERROR_MESSAGE)) {
+					continue;
+				}
+				stringBuilder.append(value);
+			}
+			failedMonitor.clear();
+			throw new ResourceNotReachableException(stringBuilder.toString());
 		}
-		if (sentRequests.get(DanteLeaderClockConstant.SYNC_INPUT).equals(false)) {
-			syncInputProperties(stats);
-			sentRequests.put(DanteLeaderClockConstant.SYNC_INPUT, true);
-			localExtendedStatistics.getStatistics().putAll(stats);
-			return Collections.singletonList(localExtendedStatistics);
+		if (!DanteLeaderClockConstant.EMPTY.equals(intervalInString)) {
+			stats.put(DanteLeaderClockConstant.NEXT_POLLING_INTERVAL, intervalInString);
 		}
-		toneGeneratorProperties(stats);
-		sentRequests.put(DanteLeaderClockConstant.TONE_GENERATOR, true);
-		localExtendedStatistics.getStatistics().putAll(stats);
-		// If we reach here then every statistic are fetched
-		ExtendedStatistics extendedStatistics = localExtendedStatistics;
-		// We initialize localExtendedStatistics and init sentRequests to fetch new statistics again (5 requests ~ 5 cpx intervals)
-		localExtendedStatistics = null;
-		initSentRequests();
-		return Collections.singletonList(extendedStatistics);
+		ExtendedStatistics extendedStatistics = new ExtendedStatistics();
+		extendedStatistics.setStatistics(stats);
+		localExtendedStatistics = extendedStatistics;
+		return Collections.singletonList(localExtendedStatistics);
 	}
 
 	/**
-	 * Initialize {@link DanteLeaderClockCommunicator#sentRequests}
+	 * Handle {@link DanteLeaderClockCommunicator#pollingInterval} adapter property
 	 */
-	private void initSentRequests() {
-		sentRequests.put(DanteLeaderClockConstant.NETWORK, false);
-		sentRequests.put(DanteLeaderClockConstant.GENERAL, false);
-		sentRequests.put(DanteLeaderClockConstant.SYNC_INPUT, false);
-		sentRequests.put(DanteLeaderClockConstant.TONE_GENERATOR, false);
-		sentRequests.put(DanteLeaderClockConstant.SYSTEM, false);
+	private long handlePollingInterval() {
```

Comment:
This could be a part of setPollingInterval() method, so the value of pollingInterval property is never below 60000 and no additional method call is needed

Reply:
Updated by moving this to setPollingInterval()